base/libgit2/strarray.jl

```diff
@@ -2,7 +2,11 @@
 
 function StrArrayStruct{T<:AbstractString}(strs::T...)
     strcount = length(strs)
-    map(s->Base.unsafe_convert(Cstring, String(s)), strs) # check for null-strings
+    for s in strs
+        if Base.containsnul(s)
+            throw("embedded NULs are not allowed in C strings: $(repr(s))")
+        end
+    end
```

Comment:
I see the one you just fixed, and also one more in `clone` in libgit2.jl, which isn't great but maybe isn't completely broken either (for now).